robyn/openapi.py

```diff
@@ -0,0 +1,263 @@
+        self.openapi_spec["paths"][modified_endpoint][route_type] = path_obj
+
+    def add_subrouter_paths(self, subrouter_openapi):
+        """
+        adds the subrouter paths to main router's openapi specs
+
+        :param subrouter_openapi: the OpenAPI object of the current subrouter
+        """
+        paths = subrouter_openapi.openapi_spec["paths"]
+
+        for path in paths:
+            self.openapi_spec["paths"][path] = paths[path]
+
+    def get_path_obj(self, endpoint: str, summary: str, tags: list, query_params: TypedDict = None):
+        """
+        get the "path" openapi object according to spec
+
+        :param endpoint: the endpoint to be added
+        :param summary: short summary of the endpoint (to be fetched from the endpoint defenition by default)
+        :param tags: tags -- for grouping of endpoints
+        :param query_params: query params for the function
+
+        :return: the "path" openapi object according to spec
+        """
+        # for converting path param from `:param` to `{param}`
+        # initialized with endpoint for handling endpoints without path params
+        endpoint_with_path_params_wrapped_in_braces = endpoint
+
+        path_params = endpoint.split(":")
+        openapi_parameter_object = []
+
+        if len(path_params) > 1:
+            path = path_params[0]
+
+            endpoint_with_path_params_wrapped_in_braces = path.removesuffix("/")
+
+            for param in path_params[1:]:
+                param_name = param[:-1] if param.endswith("/") else param
+
+                openapi_parameter_object.append(
+                    {
+                        "name": param_name,
+                        "in": "path",
+                        "required": True,
+                        "schema": {"type": "string"},
+                    }
+                )
+                endpoint_with_path_params_wrapped_in_braces += "/{" + param_name + "}"
+
+        if query_params:
+            for query_param in query_params.__annotations__:
+                # ugly hack!
+                # returns "<class 'int'>" -- this line strips out the type a.k.a int from it
+                param_type = re.findall("'[a-z]+'", str(query_params.__annotations__[query_param]))[0].replace("'", "")
+
+                openapi_parameter_object.append(
+                    {
+                        "name": query_param,
+                        "in": "query",
+                        "required": False,
+                        "schema": {"type": param_type},
+                    }
+                )
+
+        return endpoint_with_path_params_wrapped_in_braces, {
+            "summary": summary,
+            "tags": tags,
+            "parameters": openapi_parameter_object,
+            "responses": {
+                "200": {
+                    "description": "Successful Response",
+                }
+            },
+        }
+
+    def docs_handler(self):
+        """
+        get the swagger html page
```

Comment:
Also, this description is not accurate for the docs_handler. 
> get the swagger HTML page

Get where? Why is it named a hander inside the OpenAPI class?

Reply:
i have rephrased the docstrings anyways